OpenRA.Game/Server/Server.cs

```diff
@@ -627,13 +627,14 @@ byte[] CreateFrame(int client, int frame, byte[] data)
 			return ms.GetBuffer();
 		}
 
-		byte[] CreateAckFrame(int frame)
+		byte[] CreateAckFrame(int frame, byte count)
```

Comment:
`static`?

Reply:
Perhaps move a couple of these utility functions to a helper class? To keep Connection readable - leave only real logic in Connection.

These parse and packet create functions - once they work you typically don't want to read them ever again.

Reply:
You could also prepare this as an array once - perhaps wrap it inside a small class/object to abstract it away - and each time only set/update the count?  Optionally clone the array if you need need to maintain them.

Reply:
I agree, but the best solution would be to rework the server parsing loop so that it can reuse the client `OrderIO` helpers. This isn't trivial, so i'd prefer that this PR keeps our existing conventions and leave this to a future PR.